pkg/controllers/statusbackreporter/controller_integration_test.go

```diff
@@ -0,0 +1,448 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusbackreporter
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	placementv1beta1 "github.com/kubefleet-dev/kubefleet/apis/placement/v1beta1"
+	"github.com/kubefleet-dev/kubefleet/pkg/utils"
+)
+
+const (
+	// The linter in use mistakenly recognizes some of the names as potential hardcoded credentials;
+	// as a result, gosec linter warnings are suppressed for these variables.
+	crpWorkNameTemplate = "%s-work-%s" //nolint:gosec
+	nsNameTemplate      = "ns-%s"
+	crpNameTemplate     = "crp-%s"
+
+	deployName = "app"
+
+	workOrManifestAppliedReason  = "MarkedAsApplied"
+	workOrManifestAppliedMessage = "the object is marked as applied"
+	deployAvailableReason        = "MarkedAsAvailable"
+	deployAvailableMessage       = "the object is marked as available"
+)
+
+const (
+	eventuallyDuration = time.Second * 10
+	eventuallyInterval = time.Second * 1
+)
+
+var (
+	nsTemplate = corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Namespace",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nsName,
+		},
+	}
+)
+
+// createWorkObject creates a new Work object with the given name, manifests, and apply strategy.
+func createWorkObject(workName, memberClusterReservedNSName, placementObjName string, reportBackStrategy *placementv1beta1.ReportBackStrategy, rawManifestJSON ...[]byte) {
+	manifests := make([]placementv1beta1.Manifest, len(rawManifestJSON))
+	for idx := range rawManifestJSON {
+		manifests[idx] = placementv1beta1.Manifest{
+			RawExtension: runtime.RawExtension{
+				Raw: rawManifestJSON[idx],
+			},
+		}
+	}
+
+	work := &placementv1beta1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      workName,
+			Namespace: memberClusterReservedNSName,
+			Labels: map[string]string{
+				placementv1beta1.PlacementTrackingLabel: placementObjName,
+			},
+		},
+		Spec: placementv1beta1.WorkSpec{
+			Workload: placementv1beta1.WorkloadTemplate{
+				Manifests: manifests,
+			},
+			ReportBackStrategy: reportBackStrategy,
+		},
+	}
+	Expect(hubClient.Create(ctx, work)).To(Succeed())
+}
+
+func marshalK8sObjJSON(obj runtime.Object) []byte {
```

Comment:
The exact function exists in workapplier/controller_integration_test.go. Shall we consolidate?